In [1]:
import pandas as pd

# Import my csv file
file = "Resources\purchase_data.csv"
purchase_df = pd.read_csv(file)

In [2]:
purchase_df = purchase_df.rename(columns = {"Item Name" : "Item", "Purchase ID" : "ID"})
purchase_df.head()

ID             SN  Age Gender  Item ID  \
0   0        Lisim78   20   Male      108   
1   1    Lisovynya38   40   Male      143   
2   2     Ithergue48   24   Male       92   
3   3  Chamassasya86   24   Male      100   
4   4      Iskosia90   23   Male      131   

                                        Item  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

## Player Count

In [3]:
# Look at unique values in the SN column and finds the length of this array of unique values
Total_Players = len(purchase_df.SN.unique())
Total_Players

576

In [4]:
# Creates a DataFrame by using the number above and setting our index to just 0
pcount_df = pd.DataFrame(data={"Total Players" : Total_Players},index=[0])
pcount_df

Total Players
0            576

## Purchasing Analysis (Total)

In [5]:
purchase_df.head()

ID             SN  Age Gender  Item ID  \
0   0        Lisim78   20   Male      108   
1   1    Lisovynya38   40   Male      143   
2   2     Ithergue48   24   Male       92   
3   3  Chamassasya86   24   Male      100   
4   4      Iskosia90   23   Male      131   

                                        Item  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [6]:
# Takes the unique values in Item ID
Uni_item = len(purchase_df["Item ID"].unique())
Uni_item

183

In [7]:
# Sums up all the prices in the Price column and divide it by total items
Average_Price=purchase_df.Price.sum()/len(purchase_df.Price)
Average_Price

3.0509871794871795

In [8]:
# Finds the number of purchases regardless if there are dupes
Total_Purchases = len(purchase_df.ID.unique())
Total_Purchases

780

In [9]:
# Finds the sum of our Price column
Total_Revenue = purchase_df.Price.sum()
Total_Revenue

2379.77

In [10]:
# Creates a dataframe of the values above and format my columns that would need to be represented as money
pur_anal_df = pd.DataFrame(data={"Number of Unique Items":Uni_item,
                                 "Average Price":Average_Price,
                                 "Number of Purchases":Total_Purchases,
                                 "Total Revenue":Total_Revenue},index=[0])
pur_anal_df = pur_anal_df.style.format({"Average Price" : "${0:,.2f}", "Total Revenue" : '${0:,.2f}'})
pur_anal_df

## Gender Demographics

In [11]:
purchase_df.head()

ID             SN  Age Gender  Item ID  \
0   0        Lisim78   20   Male      108   
1   1    Lisovynya38   40   Male      143   
2   2     Ithergue48   24   Male       92   
3   3  Chamassasya86   24   Male      100   
4   4      Iskosia90   23   Male      131   

                                        Item  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [12]:
purchase_df.Gender.unique()

array(['Male', 'Other / Non-Disclosed', 'Female'], dtype=object)

In [13]:
# Allows for the data to only have one username per gender
gender_demo_df = purchase_df.drop_duplicates("SN")

In [14]:
# Pulls the counts of the specified index(Gender) from unique players
male = gender_demo_df.Gender.value_counts()[0]
female = gender_demo_df.Gender.value_counts()[1]
other = gender_demo_df.Gender.value_counts()[2]

In [15]:
# Finds the percentage of gender to the total count of unique players
per_male = male/gender_demo_df.ID.count()
per_female = female/gender_demo_df.ID.count()
per_other = other/gender_demo_df.ID.count()

In [16]:
# Creates a dataframe of the variables above and format my columns that would need to be represented as percentages
master_gender_demo_df = pd.DataFrame(data={"Total Count":[male,female,other],
                                           "Percentage of Players" : [per_male,per_female,per_other]},
                                    index=["Male", "Female", "Other"])
master_gender_demo_df = master_gender_demo_df.style.format({"Percentage of Players" : "{:.2%}"})
master_gender_demo_df

## Purchasing Analysis (Gender)

In [17]:
# Assign counts of gender to these variables regardless of dupes of usernames
pa_male = purchase_df.Gender.value_counts()[0]
pa_female = purchase_df.Gender.value_counts()[1]
pa_other = purchase_df.Gender.value_counts()[2]

In [18]:
# Sums the prices in the Price column grouped by Gender
gender_df = purchase_df.groupby(by=["Gender"])
gender_df.Price.sum().head()

Gender
Female                    361.94
Male                     1967.64
Other / Non-Disclosed      50.19
Name: Price, dtype: float64

In [19]:
# Assigns the values above to variables
price_m = gender_df.Price.sum()[1]
price_f = gender_df.Price.sum()[0]
price_o = gender_df.Price.sum()[2]

In [20]:
# Calculate the average purchase price by gender
paver_m = price_m/pa_male
paver_f = price_f/pa_female
paver_o = price_o/pa_other

In [21]:
# Calculates the avg total purcahse per person by gender
avert_m = price_m/male
avert_f = price_f/female
avert_o = price_o/other

In [22]:
# Creates a dataframe of the variables above and format my columns that would need to be represented as money
panal_gender_df = pd.DataFrame({"Purchase Count" : [pa_female,pa_male,pa_other], 
                                "Average Purchase Price" : [paver_f,paver_m,paver_o],
                                "Total Purchase Value" : [price_f,price_m,price_o], 
                                "Avg Total Purchase per Person": [avert_f,avert_m,avert_o]},
                               index=["Female","Male","Other"])
panal_gender_df.index.name = "Gender"
panal_gender_df = panal_gender_df.style.format({"Average Purchase Price": '${0:,.2f}',
                                               "Total Purchase Value": '${0:,.2f}',
                                               "Avg Total Purchase per Person": '${0:,.2f}'})
panal_gender_df

## Age Demographics

In [23]:
# Create our bins
bins = [0,9,14,19,24,29,34,39,100]
# Creates a list of names for our bins
age_range = ["<10","10-14","15-19","20-24","25-29","30-34","35-39","40+"]

# Creates a column to represent what the row belongs to what bins
purchase_df['AgeGroup'] = pd.cut(purchase_df['Age'],bins,labels=age_range)
purchase_df.head()

ID             SN  Age Gender  Item ID  \
0   0        Lisim78   20   Male      108   
1   1    Lisovynya38   40   Male      143   
2   2     Ithergue48   24   Male       92   
3   3  Chamassasya86   24   Male      100   
4   4      Iskosia90   23   Male      131   

                                        Item  Price AgeGroup  
0  Extraction, Quickblade Of Trembling Hands   3.53    20-24  
1                          Frenzied Scimitar   1.56      40+  
2                               Final Critic   4.88    20-24  
3                                Blindscythe   3.27    20-24  
4                                       Fury   1.44    20-24

In [24]:
# Finds the unique counts of each columns grouped by age group
woot = purchase_df.groupby(["AgeGroup"])
woot = woot.nunique()
woot

ID   SN  Age  Gender  Item ID  Item  Price  AgeGroup
AgeGroup                                                       
<10        23   17    3       2       22    21     21         1
10-14      28   22    5       2       24    24     24         1
15-19     136  107    5       3       96    96     83         1
20-24     365  258    5       3      166   163    135         1
25-29     101   77    5       3       78    78     73         1
30-34      73   52    5       2       60    60     56         1
35-39      41   31    5       3       37    37     34         1
40+        13   12    6       2       13    13     13         1

In [25]:
# Assigns Total_players to the array of value in the SN column of our dataframe called woot
Total_players = woot.SN.sum()

# Calculates the percentages of people in the game by use the values in the SN column divided by the total amount of players
Percentage_people = woot.SN/Total_players
age_df = pd.DataFrame(data={"Total Count" : woot.SN, "Percentage of Players" : Percentage_people})
age_df = age_df.style.format({"Percentage of Players" : "{:.2%}"})
age_df

## Purchasing Analysis (Age)

In [26]:
# Creates DataFrame groupby age group
age_df = purchase_df.groupby(["AgeGroup"],as_index=False)

# Calculates the amount of counts in each column
age_count_df = age_df.count()
age_count_df

AgeGroup   ID   SN  Age  Gender  Item ID  Item  Price
0      <10   23   23   23      23       23    23     23
1    10-14   28   28   28      28       28    28     28
2    15-19  136  136  136     136      136   136    136
3    20-24  365  365  365     365      365   365    365
4    25-29  101  101  101     101      101   101    101
5    30-34   73   73   73      73       73    73     73
6    35-39   41   41   41      41       41    41     41
7      40+   13   13   13      13       13    13     13

In [27]:
# Sums up the Price column of our dataframe
age_df.Price.sum()

AgeGroup    Price
0      <10    77.13
1    10-14    82.78
2    15-19   412.89
3    20-24  1114.06
4    25-29   293.00
5    30-34   214.00
6    35-39   147.67
7      40+    38.24

In [28]:
#Total Price per Age Group
psum_l10 = age_df.Price.sum().iloc[0,1]
psum_1014 = age_df.Price.sum().iloc[1,1]
psum_1519 = age_df.Price.sum().iloc[2,1]
psum_2024 = age_df.Price.sum().iloc[3,1]
psum_2529 = age_df.Price.sum().iloc[4,1]
psum_3034 = age_df.Price.sum().iloc[5,1]
psum_3539 = age_df.Price.sum().iloc[6,1]
psum_g40 = age_df.Price.sum().iloc[7,1]
agedemo_tp = [psum_l10,psum_1014,psum_1519,psum_2024,psum_2529,psum_3034,psum_3539,psum_g40]
agedemo_tp

[77.13,
 82.77999999999999,
 412.8899999999998,
 1114.0600000000006,
 292.99999999999983,
 214.0,
 147.67,
 38.24]

In [29]:
#Average Purchase Price
gave0 = psum_l10/age_count_df.iloc[0,2]
gave1 = psum_1014/age_count_df.iloc[1,2]
gave2 = psum_1519/age_count_df.iloc[2,2]
gave3 = psum_2024/age_count_df.iloc[3,2]
gave4 = psum_2529/age_count_df.iloc[4,2]
gave5 = psum_3034/age_count_df.iloc[5,2]
gave6 = psum_3539/age_count_df.iloc[6,2]
gave7 = psum_g40/age_count_df.iloc[7,2]
agedemo_avepp = [gave0,gave1,gave2,gave3,gave4,gave5,gave6,gave7]
agedemo_avepp

[3.353478260869565,
 2.956428571428571,
 3.03595588235294,
 3.0522191780821935,
 2.900990099009899,
 2.9315068493150687,
 3.6017073170731706,
 2.9415384615384617]

In [30]:
woot = purchase_df.groupby(["AgeGroup"])
woot = woot.nunique()
woot

ID   SN  Age  Gender  Item ID  Item  Price  AgeGroup
AgeGroup                                                       
<10        23   17    3       2       22    21     21         1
10-14      28   22    5       2       24    24     24         1
15-19     136  107    5       3       96    96     83         1
20-24     365  258    5       3      166   163    135         1
25-29     101   77    5       3       78    78     73         1
30-34      73   52    5       2       60    60     56         1
35-39      41   31    5       3       37    37     34         1
40+        13   12    6       2       13    13     13         1

In [31]:
# Unique count of SN gives us the number of unique people in the data
avgtpperp = agedemo_tp / woot.iloc[0:,1]
avgtpperp

AgeGroup
<10      4.537059
10-14    3.762727
15-19    3.858785
20-24    4.318062
25-29    3.805195
30-34    4.115385
35-39    4.763548
40+      3.186667
Name: SN, dtype: float64

In [32]:
agedemo_df = pd.DataFrame(data={"Average Purchase Price" : agedemo_avepp, 
                                "Total Purchase Value" : agedemo_tp, 
                                "Avg Total Purchase per Person" : avgtpperp})
agedemo_df

Average Purchase Price  Total Purchase Value  \
AgeGroup                                                 
<10                     3.353478                 77.13   
10-14                   2.956429                 82.78   
15-19                   3.035956                412.89   
20-24                   3.052219               1114.06   
25-29                   2.900990                293.00   
30-34                   2.931507                214.00   
35-39                   3.601707                147.67   
40+                     2.941538                 38.24   

          Avg Total Purchase per Person  
AgeGroup                                 
<10                            4.537059  
10-14                          3.762727  
15-19                          3.858785  
20-24                          4.318062  
25-29                          3.805195  
30-34                          4.115385  
35-39                          4.763548  
40+                            3.186667

In [33]:
# Merge our two dataframes age_count_df and agedemo on the AgeGroup column
masterage_df = pd.merge(age_count_df,agedemo_df,on="AgeGroup")

# Drops the columns we don't need
masterage_df = masterage_df.drop(columns = ["SN","Age","Gender","Item ID","Item","Price"])

# Rename the ID column to Purhcase Count
masterage_df = masterage_df.rename(columns={"ID" : "Purchase Count"})

# Set our indices to our array called AgeGroup
masterage_df = masterage_df.set_index("AgeGroup")

# Formats our dataframe to represent columns that are money with dollar signs and rounded to the second decimal place
masterage_df = masterage_df.style.format({"Average Purchase Price": '${0:,.2f}',
                                          "Total Purchase Value": '${0:,.2f}',
                                          "Avg Total Purchase per Person": '${0:,.2f}' })
masterage_df

## Top Spenders

In [34]:
# Group by the usernames in our orignal dataframe
top = purchase_df.groupby(["SN"])

# Sums up the price column in our dataframe grouped by usernames
top_Price = top.Price.sum()

In [35]:
# Sort our values in the Price column with from greatest to least
top_Price = top_Price.sort_values(ascending=False)

# Calls upon the top 5 rows and the names of these indices
top_Price.index[0:5]

Index(['Lisosia93', 'Idastidru52', 'Chamjask73', 'Iral74', 'Iskadarya95'], dtype='object', name='SN')

In [36]:
# Create an array of these names
top_5 = ['Lisosia93', 'Idastidru52', 'Chamjask73', 'Iral74', 'Iskadarya95']

In [37]:
# For loop to append the count of the names in top_5 and append them to test
test = []
for x in top_5:
    test.append(purchase_df.loc[purchase_df.SN == x ].count())
top_count = pd.DataFrame(test, index=top_5)
top_count["ID"]

Lisosia93      5
Idastidru52    4
Chamjask73     3
Iral74         4
Iskadarya95    3
Name: ID, dtype: int64

In [38]:
# For loop to find the sum of the purchases based on each name in top_5
testy = []
for x in top_5:
    testy.append(purchase_df.loc[purchase_df.SN == x ].sum())
top_price = pd.DataFrame(testy, index=top_5)
top_price["Price"]

Lisosia93      18.96
Idastidru52    15.45
Chamjask73     13.83
Iral74         13.62
Iskadarya95    13.10
Name: Price, dtype: float64

In [39]:
# For loop to append to our empty list of test1 of the calculations of average price per person
test1 = []
for x in range(0,5):
    test1.append(top_price["Price"][x]/top_count["ID"][x])
top_ave = pd.DataFrame(test1, index=top_5, columns=["Average Price"])
top_ave

Average Price
Lisosia93         3.792000
Idastidru52       3.862500
Chamjask73        4.610000
Iral74            3.405000
Iskadarya95       4.366667

In [40]:
# Creates a dataframe of the variables above and format my columns that would need to be represented as money
top_spend_df = pd.DataFrame(data={"Purchase Count" : top_count["ID"], 
                   "Average Purchase Price" : top_ave["Average Price"], 
                   "Total Purchase Value" : top_price["Price"]}, index=top_5)
top_spend_df = top_spend_df.style.format({"Average Purchase Price" : '${0:,.2f}', "Total Purchase Value" : '${0:,.2f}'})
top_spend_df

## Most Popular Items

In [41]:
purchase_df.head()

ID             SN  Age Gender  Item ID  \
0   0        Lisim78   20   Male      108   
1   1    Lisovynya38   40   Male      143   
2   2     Ithergue48   24   Male       92   
3   3  Chamassasya86   24   Male      100   
4   4      Iskosia90   23   Male      131   

                                        Item  Price AgeGroup  
0  Extraction, Quickblade Of Trembling Hands   3.53    20-24  
1                          Frenzied Scimitar   1.56      40+  
2                               Final Critic   4.88    20-24  
3                                Blindscythe   3.27    20-24  
4                                       Fury   1.44    20-24

In [42]:
# Assign our orignal dataframe to another variable
og_pop_df = purchase_df

# Assigns the values in Item ID to the indices of our dataframe
index_pop_df = purchase_df.set_index("Item ID")

# Creates a dataframe groupby item, item id and price and does NOT set them as our indices and then sorts by ID or purchase counts
pop_df = og_pop_df.groupby(["Item","Item ID", "Price"], as_index = False).count().sort_values(by="ID",ascending=False)
pop_df.head()

Item  Item ID  Price  ID  SN  \
95   Oathbreaker, Last Hope of the Breaking Storm      178   4.23  12  12   
52      Extraction, Quickblade Of Trembling Hands      108   3.53   9   9   
94                                        Nirvana       82   4.90   9   9   
56                           Fiery Glass Crusader      145   4.58   9   9   
108                 Pursuit, Cudgel of Necromancy       19   1.02   8   8   

     Age  Gender  AgeGroup  
95    12      12        12  
52     9       9         9  
94     9       9         9  
56     9       9         9  
108    8       8         8

In [43]:
# Based on these item ids which are from the top 5 items based on purchase count, return the item and price values sum together
pop_price = index_pop_df.loc[[178,145,108,82,19],
                             ["Item","Price"]].groupby(["Item ID","Item"]).sum().sort_values(by="Price",ascending=False)
# Rename our Price column to total price
pop_price = pop_price.rename(columns={"Price" :"Total Price"})
pop_price

Total Price
Item ID Item                                                     
178     Oathbreaker, Last Hope of the Breaking Storm        50.76
82      Nirvana                                             44.10
145     Fiery Glass Crusader                                41.22
108     Extraction, Quickblade Of Trembling Hands           31.77
19      Pursuit, Cudgel of Necromancy                        8.16

In [44]:
# Merge our two dataframes on the Item names and keep all the columns together
pop_item = pd.merge(pop_price,pop_df,on=("Item"),how="inner")
pop_item

Item  Total Price  Item ID  Price  \
0  Oathbreaker, Last Hope of the Breaking Storm        50.76      178   4.23   
1                                       Nirvana        44.10       82   4.90   
2                          Fiery Glass Crusader        41.22      145   4.58   
3     Extraction, Quickblade Of Trembling Hands        31.77      108   3.53   
4                 Pursuit, Cudgel of Necromancy         8.16       19   1.02   

   ID  SN  Age  Gender  AgeGroup  
0  12  12   12      12        12  
1   9   9    9       9         9  
2   9   9    9       9         9  
3   9   9    9       9         9  
4   8   8    8       8         8

In [45]:
# Drop our columns we don't need
pop_item = pop_item.drop(columns=["SN","Age","Gender","AgeGroup"])
pop_item

Item  Total Price  Item ID  Price  \
0  Oathbreaker, Last Hope of the Breaking Storm        50.76      178   4.23   
1                                       Nirvana        44.10       82   4.90   
2                          Fiery Glass Crusader        41.22      145   4.58   
3     Extraction, Quickblade Of Trembling Hands        31.77      108   3.53   
4                 Pursuit, Cudgel of Necromancy         8.16       19   1.02   

   ID  
0  12  
1   9  
2   9  
3   9  
4   8

In [46]:
# Rename our ID column to Count
pop_item = pop_item.rename(columns = {"ID" : "Count"})

# Set our indices to the values in Item ID
pop_item = pop_item.set_index("Item ID")

# Formats our dataframe by formating our columns that are money with dollar signs and rounded to the second decimal
pop_item = pop_item.style.format({"Total Price" : '${0:,.2f}', "Price":'${0:,.2f}'})
pop_item

## Most Profitable Items

In [47]:
# Group by item and item ID and does NOT set them as our indices, sums the prices of these items and sorts them by Greatest to least
pro_items = og_pop_df.groupby(["Item","Item ID"], as_index = False).sum().sort_values(by="Price",ascending=False)

# Rename our column Price to Total Price
pro_items = pro_items.rename(columns={"Price" : "Total Price"})

# Drop the Columns we do not want
pro_items = pro_items.drop(columns = ["ID","Age"])
pro_items.head()

Item  Item ID  Total Price
95   Oathbreaker, Last Hope of the Breaking Storm      178        50.76
94                                        Nirvana       82        44.10
56                           Fiery Glass Crusader      145        41.22
57                                   Final Critic       92        39.04
128                                Singed Scalpel      103        34.80

In [48]:
# Creates a dataframe called test which calculated the total count of each item
test = purchase_df.groupby(["Item","Item ID", "Price"], as_index = False).count().sort_values(by="ID",ascending=False)
test.head()

Item  Item ID  Price  ID  SN  \
95   Oathbreaker, Last Hope of the Breaking Storm      178   4.23  12  12   
52      Extraction, Quickblade Of Trembling Hands      108   3.53   9   9   
94                                        Nirvana       82   4.90   9   9   
56                           Fiery Glass Crusader      145   4.58   9   9   
108                 Pursuit, Cudgel of Necromancy       19   1.02   8   8   

     Age  Gender  AgeGroup  
95    12      12        12  
52     9       9         9  
94     9       9         9  
56     9       9         9  
108    8       8         8

In [49]:
# Drop the columns we do not want
test = test.drop(columns=["SN","Age","Gender","AgeGroup"])
test = test.rename(columns={"ID" : "Purchase Count"})
test.head()

Item  Item ID  Price  \
95   Oathbreaker, Last Hope of the Breaking Storm      178   4.23   
52      Extraction, Quickblade Of Trembling Hands      108   3.53   
94                                        Nirvana       82   4.90   
56                           Fiery Glass Crusader      145   4.58   
108                 Pursuit, Cudgel of Necromancy       19   1.02   

     Purchase Count  
95               12  
52                9  
94                9  
56                9  
108               8

In [50]:
# Merge our two dataframes together
top_pro_item = pd.merge(test,pro_items,on="Item ID")
top_pro_item.head()

Item_x  Item ID  Price  \
0  Oathbreaker, Last Hope of the Breaking Storm      178   4.23   
1     Extraction, Quickblade Of Trembling Hands      108   3.53   
2                                       Nirvana       82   4.90   
3                          Fiery Glass Crusader      145   4.58   
4                 Pursuit, Cudgel of Necromancy       19   1.02   

   Purchase Count                                        Item_y  Total Price  
0              12  Oathbreaker, Last Hope of the Breaking Storm        50.76  
1               9     Extraction, Quickblade Of Trembling Hands        31.77  
2               9                                       Nirvana        44.10  
3               9                          Fiery Glass Crusader        41.22  
4               8                 Pursuit, Cudgel of Necromancy         8.16

In [51]:
# Drop the identical column
top_pro_item = top_pro_item.drop(columns = ["Item_y"])

# Renames our column
top_pro_item = top_pro_item.rename(columns={"Item_x" : "Item"})

# Sort our values by the Purchase Count
top_pro_item = top_pro_item.sort_values(by="Purchase Count",ascending=False)

# Set our indices by the item id
top_pro_item = top_pro_item.set_index("Item ID")
top_pro_item.head()

Item  Price  Purchase Count  \
Item ID                                                                        
178      Oathbreaker, Last Hope of the Breaking Storm   4.23              12   
82                                            Nirvana   4.90               9   
145                              Fiery Glass Crusader   4.58               9   
108         Extraction, Quickblade Of Trembling Hands   3.53               9   
92                                       Final Critic   4.88               8   

         Total Price  
Item ID               
178            50.76  
82             44.10  
145            41.22  
108            31.77  
92             39.04

In [52]:
# Takes the top 5 rows and assign them to the different name
top5_pro_item = top_pro_item[:5]

# Formats our dataframe
top5_pro_item = top5_pro_item.style.format({"Price":"${0:,.2f}","Total Price":"${0:,.2f}"})
top5_pro_item

# Data Frames

In [53]:
# Player Count
pcount_df

Total Players
0            576

In [54]:
# Purchasing Analysis (Total)
pur_anal_df

In [55]:
# Gender Demographics
master_gender_demo_df

In [56]:
# Purchasing Analysis (Gender)
panal_gender_df

In [57]:
# Age Demographics
age_df

In [58]:
# Purchasing Analysis (Age)
masterage_df

In [59]:
# Top Spenders
top_spend_df

In [60]:
# Most Popular Items
pop_item

In [61]:
# Most Profitable Items
top5_pro_item

# Observations

My first observation I notice is the age group of the playerbase for this game. The majority of the players are from the age of 20 to 24 which accounts for 44.79% of the total population.

My second observation I notice is that the purchases from this age group of 20 to 24 accounts for about 47% of the total revenue for the game.

My last observation I notice is that males are more incline to spend money for this game since males have a purchase count of 652 out of 780 purchases.